api/src/main/java/org/openmrs/module/pihcore/PihCoreActivator.java

```diff
@@ -220,6 +222,48 @@ public void setGlobalProperties(Config config) {
         }
     }
 
+    public User setupCommCareUser(){
+
+	    User user = null;
+        String commCareUserName = Context.getRuntimeProperties().getProperty("commcare.username", "commCareUser");
+        String commCareUserPassword = Context.getRuntimeProperties().getProperty("commcare.password", "CommCare12");
+        if (StringUtils.isNotBlank(commCareUserName)) {
+            user = Context.getUserService().getUserByUsername(commCareUserName);
+            if (user == null) {
+                // create a new user if one does not already exist
+                user = new User();
+            }
+            user.setUsername(commCareUserName);
+            Person person = user.getPerson();
+            if (person == null) {
+                person = new Person();
+                person.addName(new PersonName("CommCare", "", "Integration"));
+                person.setGender("M");
+                user.setPerson(person);
+            }
+
+            user.addRole(Context.getUserService().getRole("Authenticated"));
```

Comment:
What's this role, I forget? (Though from the name it does seem correct that this user should have it... :) )

Reply:
I think that basically this gives read permissions:
https://humci.pih-emr.org/mirebalais/admin/users/role.form?roleName=Authenticated